File: frontend/src/types/agent-run.ts
import { z } from "zod";

export const agentRunStatusSchema = z.enum(["running", "succeeded", "failed"]);
export const agentRunSchema = z.object({
  id: z.uuid(),
  request_id: z.string(),
  conversation_id: z.uuid().nullable(),
  ticket_id: z.uuid().nullable(),
  status: agentRunStatusSchema,
  model_name: z.string(),
  intent: z.string().nullable(),
  priority: z.string().nullable(),
  requires_human: z.boolean().nullable(),
  decision_reason: z.string().nullable(),
  tool_name: z.string().nullable(),
  ticket_code: z.string().nullable(),
  duration_ms: z.number().int().nonnegative().nullable(),
  error_type: z.string().nullable(),
  error_message: z.string().nullable(),
  started_at: z.iso.datetime({ offset: true }),
  completed_at: z.iso.datetime({ offset: true }).nullable(),
});
export const agentRunListSchema = z.object({
  items: z.array(agentRunSchema),
  total: z.number().int().nonnegative(),
  offset: z.number().int().nonnegative(),
  limit: z.number().int().positive(),
});
export const agentRunStatisticsSchema = z.object({
  total: z.number().int().nonnegative(),
  running: z.number().int().nonnegative(),
  succeeded: z.number().int().nonnegative(),
  failed: z.number().int().nonnegative(),
  tool_calls: z.number().int().nonnegative(),
  average_duration_ms: z.number().nonnegative(),
});

export type AgentRunStatus = z.infer<typeof agentRunStatusSchema>;
export type AgentRun = z.infer<typeof agentRunSchema>;
export type AgentRunList = z.infer<typeof agentRunListSchema>;
export type AgentRunStatistics = z.infer<typeof agentRunStatisticsSchema>;

export interface AgentRunFilters {
  status: AgentRunStatus | "";
  toolCalled: "" | "true" | "false";
  keyword: string;
  page: number;
  pageSize: number;
}

File: backend/app/schema/agent_run.py
"""Agent 运行记录的只读接口协议，与 Graph 内部运行状态分开维护。"""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.db.models import AgentRunStatus


class AgentRunResponse(BaseModel):
    """运行详情只暴露诊断元数据，不返回客户原始消息和密钥。"""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    request_id: str
    conversation_id: UUID | None
    ticket_id: UUID | None
    status: AgentRunStatus
    model_name: str
    intent: str | None
    priority: str | None
    requires_human: bool | None
    decision_reason: str | None
    tool_name: str | None
    ticket_code: str | None
    duration_ms: int | None
    error_type: str | None
    error_message: str | None
    started_at: datetime
    completed_at: datetime | None


class AgentRunListResponse(BaseModel):
    items: list[AgentRunResponse]
    total: int
    offset: int
    limit: int


class AgentRunStatisticsResponse(BaseModel):
    total: int
    running: int
    succeeded: int
    failed: int
    tool_calls: int
    average_duration_ms: float
